Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Pumps/CondenserWater/Subsequences/Speed.mo

```diff
@@ -0,0 +1,222 @@
+﻿within Buildings.Controls.OBC.ASHRAE.PrimarySystem.ChillerPlant.Pumps.CondenserWater.Subsequences;
+block Speed
+  "Output design speed of condenser water pumps at current stage"
+
+  parameter Boolean haveWSE = true
+    "Flag to indicate if the plant has water side economiser";
+  parameter Integer totChiSta = 6
+    "Total number of stages, include the stage like chiller stage 0 plus WSE";
+  parameter Real staVec[totChiSta] = {0, 0.5, 1, 1.5, 2, 2.5}
+    "Chiller stage vector, element value like 0.5 means stage 0 plus WSE";
+  parameter Real conWatPumSpeSet[totChiSta] = {0, 0.5, 0.75, 0.6, 0.75, 0.9}
+    "Condenser water pump speed setpoint, according to current chiller stage and WSE status"
+    annotation (Dialog(group="Setpoint according to stage"));
+  parameter Real conWatPumOnSet[totChiSta] = {0,1,1,2,2,2}
+    "Number of condenser water pumps that should be ON, according to current chiller stage and WSE status"
+    annotation (Dialog(group="Setpoint according to stage"));
+
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput uChiSta
+    "Current chiller stage"
+    annotation (Placement(transformation(extent={{-180,40},{-140,80}}),
+      iconTransformation(extent={{-140,20},{-100,60}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uWSE if haveWSE
+    "Water side economizer status: true = ON, false = OFF"
+    annotation (Placement(transformation(extent={{-180,-80},{-140,-40}}),
+      iconTransformation(extent={{-140,-60},{-100,-20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerOutput yConWatPumNum
```

Comment:
I'd prefer to create anther sequence which can be used to convert the number of operating device (along with current devices operating status) to the desired the devices operating status.
So it could be used by other devices if necessary.

Reply:
I'm fine with that approach.